File: MultipleInheritance.java
public class MultipleInheritance {
    public static void main(String[] args) {
        C c = new C();
        c.doSomething();
    }
}

interface P {
    default void doSomething() {
        System.out.println("P's doSomething()");
    }
}

interface Q {
    default void doSomething() {
        System.out.println("Q's doSomething()");
    }
}

class C implements P, Q {
    @Override
    public void doSomething() {
        P.super.doSomething();
        Q.super.doSomething();
        System.out.println("C's doSomething()");
    }
}

File: multipleInheritance.kt
interface P {
    fun doSomething() {
        println("P's doSomething()")
    }
}

interface Q {
    fun doSomething() {
        println("Q's doSomething()")
    }
}

class C : P, Q {
    override fun doSomething() {
        super<P>.doSomething()
        super<Q>.doSomething()
        println("C's doSomething()")
    }
}

fun main() {
    val c = C()
    c.doSomething()
}
